### PR TITLE
Mark SwiftUIRepresentable as Deprecated

### PR DESCRIPTION
**Pull request by adolfogarza**
### 🔗 Issue Links

https://stream-io.atlassian.net/browse/CIS-1562

### 🎯 Goal

_Mark SwiftUIRepresentable as Deprecated_

### ☑️ Contributor Checklist
- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### CODE REVIEW

File: Sources/StreamChat/Controllers/ChannelController/ChannelController+SwiftUI.swift [hunk 1]
```diff
@@ -8,6 +8,7 @@ import SwiftUI
 @available(iOS 13, *)
 extension ChatChannelController {
     /// A wrapper object that exposes the controller variables in the form of `ObservableObject` to be used in SwiftUI.
+    @available(*, deprecated, message: "This is now deprecated, please refer to the SwiftUI SDK at https://github.com/GetStream/stream-chat-swiftui")
```
[the remark this answers is not in the document]
Comment (nuno-vieira) [reply Sources/StreamChat/Controllers/ChannelController/ChannelController+SwiftUI.swift:11]: I don't think we should actually, we have a lot of customers using this and you can always use Combine with UIKit.

[the remark this answers is not in the document]
Comment (martinmitrevski) [reply Sources/StreamChat/Controllers/ChannelController/ChannelController+SwiftUI.swift:11]: Yes, exactly, that's what I was thinking too.

**Review by martinmitrevski — COMMENTED** [review]
There are still some places where controllers are unnecessarily deprecated like `ChatUserListController` along with its properties. Basically, in the LLC we shouldn't have any deprecations, only in the UI layer.

Also, I've noticed a lot of new warnings in the UIKit SDK. We need to check those too.

**Review by martinmitrevski — APPROVED** [review]
LGTM ✅

File: Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView+SwiftUI.swift [hunk 2]
```diff
@@ -9,6 +9,7 @@ import SwiftUI
 @available(iOS 13.0, *)
 /// Protocol of `ChatChannelListItemView` wrapper for use in SwiftUI.
 public protocol ChatChannelListItemViewSwiftUIView: View {
+    @available(*, deprecated, message: "This is now deprecated, please refer to the SwiftUI SDK at https://github.com/GetStream/stream-chat-swiftui")
```
Comment (nuno-vieira) [inline Sources/StreamChatUI/ChatChannelList/ChatChannelListItemView+SwiftUI.swift:12]: The deprecation flags should actually be above the ChatChannelListItemViewSwiftUIView. Any reason why they are on the initialisers?

**Review by nuno-vieira — CHANGES_REQUESTED** [review]
This should just be a question of Deprecating the SwiftUI Views, making the @available mark above the view types. There's no need to deprecate properties or functional calls

**Pull request closed without merging**